dags/common/utils.py

```diff
@@ -12,7 +12,12 @@ def set_harvesting_interval(repo, **kwargs):
             "start_date": kwargs["params"]["start_date"],
             "until_date": kwargs["params"]["until_date"],
         }
-    start_date = repo.find_the_last_uploaded_file_date()
+    start_date = (
+        "params" in kwargs
+        and "start_date" in kwargs["params"]
+        and kwargs["params"]["start_date"]
```

Comment:
It is a bit odd to me cause it will return boolean and `find_the_last_uploaded_file_date` will return a date.
Did you mean something like 
`kwargs["params"]["start_date"] if kwargs.get('params', {}).get('start_date')`?

Reply:
no, is not a boolean which is returned here.
kwargs["params"]["start_date"] is string. 
Here what I am doing is: 
1. Checking is params even in kwargs for this task. It can be that kwargs will be empty if we won't pass anything before triggering the task. How to pass it? In UI of airflow, before triggering the run (and run consists of tasks)
2. Checking if `start_date` is in kwargs['params']. We can pass whichever value there, not just `start_date`
3. If we have `params`, and `params['start_date']` defined, I return the value of `params['start_date']`. For example: 

```python 
params['start_date'] = "2020-08-08"
'start_date' in params and params['start_date'] == "2020-08-08"
```

Reply:
So I would suggest doing `kwargs.get('params', {}).get('start_date')` instead
```
        "params" in kwargs
        and "start_date" in kwargs["params"]
        and kwargs["params"]["start_date"]
```
to me, it's a bit more readable with just one statement